commit 1b1bcfc89ac08be2fc9fd5df44b98ec44e4d909b (refs/changes/27/5627/4)
Author: Robert Mustacchi <rm@joyent.com>
Date:   2019-02-26T01:00:22+00:00 (8 months ago)
    
    OS-7592 viona doesn't zero ULP checksum for HCK_FULLCKSUM
    Reviewed by: Patrick Mooney <patrick.mooney@joyent.com>
    Reviewed by: Ryan Zezeski <rpz@joyent.com>
    Approved by: Patrick Mooney <patrick.mooney@joyent.com>

diff --git a/usr/src/uts/i86pc/io/viona/viona.c b/usr/src/uts/i86pc/io/viona/viona.c
index 4ae0f7317f..74fd237fc9 100644
--- a/usr/src/uts/i86pc/io/viona/viona.c
+++ b/usr/src/uts/i86pc/io/viona/viona.c
@@ -2661,6 +2661,8 @@ viona_tx_csum(viona_vring_t *ring, const struct virtio_net_hdr *hdr,
 	if (ftype == ETHERTYPE_IP) {
 		if ((link->l_cap_csum & HCKSUM_INET_FULL_V4) != 0 &&
 		    (ipproto == IPPROTO_TCP || ipproto == IPPROTO_UDP)) {
+			uint16_t *csump = (uint16_t *)(mp->b_rptr + csum_stuff);
+			*csump = 0;
 			flags |= HCK_FULLCKSUM;
 			mac_hcksum_set(mp, 0, 0, 0, 0, flags);
 			return (B_TRUE);
@@ -2673,6 +2675,8 @@ viona_tx_csum(viona_vring_t *ring, const struct virtio_net_hdr *hdr,
 	} else if (ftype == ETHERTYPE_IPV6) {
 		if ((link->l_cap_csum & HCKSUM_INET_FULL_V6) != 0 &&
 		    (ipproto == IPPROTO_TCP || ipproto == IPPROTO_UDP)) {
+			uint16_t *csump = (uint16_t *)(mp->b_rptr + csum_stuff);
+			*csump = 0;
 			flags |= HCK_FULLCKSUM;
 			mac_hcksum_set(mp, 0, 0, 0, 0, flags);
 			return (B_TRUE);
